kcidb/monitor/subscriptions/linux_stable_rt.py

```diff
@@ -16,11 +16,19 @@ def match_revision(revision):
     if repo_url not in revision.repo_branch_checkouts:
         return ()
 
+    selected_origins = {'maestro', 'broonie'}
+    # If the revision is not from 'maestro' or 'broonie' origin
+    if not {c.origin for c in revision.checkouts} & selected_origins:
+        return ()
+
+    # Return if there are no builds or status is unknown for all the builds
     if revision.builds_valid is None:
         return ()
 
-    # If the revision is not from 'maestro' or 'broonie' origin
-    if not {c.origin for c in revision.checkouts} & {'maestro', 'broonie'}:
+    # If no builds found from 'maestro' or 'broonie' origin
+    builds = [build for build in revision.builds
+              if build.origin in selected_origins]
+    if not builds:
```

Comment:
You can replace all this with:
```suggestion
    if all(b.origin not in selected_origins or b.valid is None
           for b in revision.builds):
```
You don't really have to check the checkout origins.

Reply:
Thanks for the suggestion. This is neat.
Updated both subscription modules. Also, added a fix for fluster test report.